release/src/main/scripts/build_release_candidate.sh

```diff
@@ -175,10 +237,9 @@ if [[ $confirmation = "y" ]]; then
   cd "${LOCAL_PYTHON_STAGING_DIR}"
 
   echo '-------------------Cloning Beam Release Branch-----------------'
-  git clone "${GIT_REPO_URL}"
+  git clone --branch "${RC_TAG}" --depth 1 "${GIT_REPO_URL}"
   cd "${BEAM_ROOT_DIR}"
-  git checkout "${RELEASE_BRANCH}"
-  RELEASE_COMMIT=$(git rev-parse --verify HEAD)
+  RELEASE_COMMIT=$(git rev-list -n 1 "tags/${RC_TAG}")
```

Comment:
how come this changed? Isn't it desirable to have the RC commits be part of the release-xxx branch?

Reply:
This particular line works whether or not the RC tag is part of the release branch, so I would keep it either way.

But your question is still legit. The background is:

 - [dev@ thread](https://lists.apache.org/thread.html/205472bdaf3c2c5876533750d417c19b0d1078131a3dc04916082ce8%40%3Cdev.beam.apache.org%3E) discussing ideal git release branch/RC configuration
 - [new branch cut instructions](https://beam.apache.org/contribute/release-guide/#4-create-a-release-branch-in-apachebeam-repository) with diagram of end state and alterations to support script in #14239 (TL;DR remain on SNAPSHOT/dev version on release branch, advance the version on master branch)
 - [new RC tagging instructions](https://beam.apache.org/contribute/release-guide/#tag-a-chosen-commit-for-the-rc) with diagram of end state and support script in #14213 (TL;DR a single tagged commit off the release branch that removes SNAPSHOT/dev, instead of the same commit that is immediately reverted)
 - removal of RC tagging from building RC in #14512

And this PR is basically just bugfixes so that the script works with the new setup.

Reply:
I see I see... okay, makes sense. Thanks LGTM : )